### ISSUE TITLE
Change build.sh to apply patches to version-bumped git build

### ISSUE THREAD

**Issue by TwoLeaves** [pyroscope/rtorrent-ps#27]
Commit https://github.com/rakshasa/rtorrent/commit/30d8379391ad4cb3097d57aa56a488d061e68662 merges the "cmd_ui_current_view" from rtorrent-pryo's codebase. However since the function is still defined in rtorrent-pyro, compiling rtorrent-pyro fails with this error complaining about multiple definitions:

```
libsub_root.a(command_pyroscope.o): In function `cmd_ui_current_view()':
command_pyroscope.cc:(.text+0x190): multiple definition of `cmd_ui_current_view()'
libsub_root.a(command_ui.o):command_ui.cc:(.text+0x3b0): first defined here
collect2: error: ld returned 1 exit status
Makefile:548: recipe for target 'rtorrent' failed
make[3]: *** [rtorrent] Error 1
```

Is this a bug in rtorrent or rtorrent-pyro?

**Comment by TwoLeaves** [comment]
Looks like the function should be removed from rtorrent-ps, now that pyroscope's request to merge the "cmd_ui_current_view" function into rtorrent has been accepted.

**Comment by pyroscope** [comment]
No it shouldn't, since that 'd break release builds. The problem is Jari not bumping the version to 0.9.7 after release, which makes these conditions not work:

    #if RT_HEX_VERSION <= 0x000906

So patch the configure.in to proper dev versioning before building a git version.

**Comment by pyroscope** [comment]
Just now improving git build proces to carry proper versions.

**Comment by pyroscope** [comment]
:checkered_flag:  Status: git builds and runs as 0.9.7/0.13.7, but without patches

**Comment by chros73** [comment]
> but without patches

- you can grab all my (0.9.7/0.13.7) patches if you use the same commits that are in my build.sh.
- also you have to a look which your files (*.cc) are modifed to handle various stuff

**Comment by chros73** [comment]
Oh, I almost forgot it, 1 more:
- I also applied [Add ability in build script to use 'all' in librottent patches](https://github.com/chros73/rtorrent-ps/issues/27) 
- this `all` keyword also applies all the other patches: `misc`, `ps`, etc.

I also labelled all my issues with `build` when some important change happened in `build.sh`. 

Now, your original build.sh script is perfect! :)

**Comment by pyroscope** [comment]
When a git build is properly patched (carries a 0.9.7/0.13.7 version like it should), things do work by now (if not, reopen with a problem description).

build.sh has the code to do that patch, and now also supports multi-version tagging in patch file names (so you can do 0.9.7 only, or ``*_0.9.6_0.9.7_*``).

**Comment by chros73** [comment]
Awesome!

> and now also supports multi-version tagging in patch file names

:) I forgot to mention this one.
